src/BizHawk.Emulation.Cores/Consoles/Nintendo/QuickNES/QuickNES.cs

```diff
@@ -82,22 +82,54 @@ public QuickNES(byte[] file, QuickNESSettings settings, QuickNESSyncSettings syn
 		private void SetControllerDefinition()
 		{
 			ControllerDefinition def = new("NES Controller");
+
 			void AddButtons(IEnumerable<(string PrefixedName, uint Bitmask)> entries)
 				=> def.BoolButtons.AddRange(entries.Select(static p => p.PrefixedName));
-			AddButtons(_syncSettings.Port1 switch
+
+			if (_syncSettings.Port1 == Port1PeripheralOption.Gamepad || _syncSettings.Port1 == Port1PeripheralOption.FourScore || _syncSettings.Port1 == Port1PeripheralOption.ArkanoidFamicom)

```

Comment:
This condition would be much shorter with pattern matching, but the whole thing is actually redundant with the `switch`.